testsuite/integration-arquillian/tests/pom.xml

```diff
@@ -125,7 +125,7 @@
         <app.server.2.debug.suspend>n</app.server.2.debug.suspend>
         <app.server.2.jboss.jvm.debug.args>-agentlib:jdwp=transport=dt_socket,server=y,suspend=${app.server.2.debug.suspend},address=localhost:${app.server.2.debug.port}</app.server.2.jboss.jvm.debug.args>
         <app.server.memory.Xms>64m</app.server.memory.Xms>
-        <app.server.memory.Xmx>512m</app.server.memory.Xmx>
+        <app.server.memory.Xmx>768m</app.server.memory.Xmx>
```

Comment:
Not sure if this change is safe. Seems like newer versions of BC are more memory demanding (see a [failing test](https://github.com/keycloak/keycloak/actions/runs/5507937506/jobs/10048118779) without this change). Is that something we want to risk as it might be directly affecting users.

Reply:
That's good point. Hopefully it affects just the app-server (which may not be that important AFAIK due the adapter tests executed on embedded undertow by default?).

Isn't it possible that app-server has both jdk15on and jdk18on dependencies and that being the cause of those issues?

Reply:
We run with `app-server-wildfly` [by default](https://github.com/keycloak/keycloak/blob/67b20dfd9b7c63f96bbd81a6ce8562f663fde56c/testsuite/integration-arquillian/tests/base/pom.xml#L521).

I did a little bit more digging into this. I realized we're actually not including any BC dependencies with the WF/EAP adapter. We [use](https://github.com/keycloak/keycloak/blob/67b20dfd9b7c63f96bbd81a6ce8562f663fde56c/distribution/feature-packs/adapter-feature-pack/src/main/resources/modules/system/add-ons/keycloak/org/keycloak/keycloak-crypto-default/main/module.xml#L33) the default BC module that's already present in WF. You can simply double check that by building the adapters dist and checking the JARs included. There are actually no 3rd party deps bundled in our adapters dist. That means WF/EAP adapter should not be affected by this change at all.

The reason for the out of memory test failures is that we're including the BC deps in our test app WARs. The newer versions of BC libs are a few MBs larger. Hence I believe we're reaching the memory limit because of this.

However, where we are also including the BC deps is the Tomcat and Jetty adapters. Those are affected by this upgrade. This is NOT tested in GHA. @mposolda @miquelsi Do we need to run the full internal pipeline to test this?

Reply:
Running internal adapter pipeline can be nice, however our java adapters are deprecated. So not 100% sure it is strictly needed if it is too much work... IMO any performance/memory "regression" specific only to java adapters is fine due their deprecation (as long as it does not affect the Keycloak server).

Reply:
@mposolda Are you then fine with merging this? :) It'd be nice to have it in 22.0.1 to fix https://github.com/keycloak/keycloak/issues/21664

Reply:
Yes, ok. I am approving